File: lps.scala
import scala.util.control.Breaks.{ breakable, break }

object lps {
  def time_usec( time: Long ): Long = {
    time * 1000
  }
  
  def main( args: Array[ String ] ){
    var argc = args.size
    var sec = 1
    if ( 1 <= argc ){
      sec = Integer.parseInt( args( 0 ) )
    }
    var usec = sec * 1000000
    
    var count = 0
    var start_usec = time_usec( System.currentTimeMillis() )
    breakable {
      while ( true ){
        count += 1
        var end_usec = time_usec( System.currentTimeMillis() )
        var update_usec = end_usec - start_usec
        if ( usec <= update_usec ){
          printf( "count=%d update_usec=%d\n", count, update_usec )
          break
        }
      }
    }
  }
}

File: lps.php
<?php
function time_usec( $time ){
  return $time * 1000000;
}

$sec = isset( $argv[ 1 ] ) ? intval( $argv[ 1 ] ) : 1;
if ( $sec <= 0 ){
  $sec = 1;
}
$usec = $sec * 1000000;

$count = 0;
$start_usec = time_usec( microtime( true ) );
while ( true ){
  $count += 1;
  $end_usec = time_usec( microtime( true ) );
  $update_usec = $end_usec - $start_usec;
  if ( $usec <= $update_usec ){
    printf( "count=%u update_usec=%u\n", $count, $update_usec );
    break;
  }
}
